### PR TITLE
#36: Allow Openshift url to be specified via system properties or env…

### PR DESCRIPTION
**Pull request by iocanel**
… variable (if not explicitly specified in config). When using custom Openshift Url the adapter should not consult root paths.

### CODE REVIEW

File: openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftConfig.java [hunk 1]
```diff
@@ -58,15 +61,22 @@ public OpenshiftConfig(Config kubernetesConfig, String oapiVersion, String openS
     this.oapiVersion = Utils.getSystemPropertyOrEnvVar(KUBERNETES_OAPI_VERSION_SYSTEM_PROPERTY, oapiVersion);
     this.openShiftUrl = openShiftUrl;
 
-    if (this.openShiftUrl == null) {
-      try {
-        this.openShiftUrl = new URL(new URL(getMasterUrl()), "/oapi/" + this.oapiVersion + "/").toString();
-      } catch (MalformedURLException e) {
-        throw KubernetesClientException.launderThrowable(e);
+    if (this.openShiftUrl == null || this.openShiftUrl.isEmpty()) {
+      this.openShiftUrl = getCutsomOpenshiftUrl();
```
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftConfig.java]: typo: cutsom

File: openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftConfig.java [hunk 2]
```diff
@@ -58,15 +61,22 @@ public OpenshiftConfig(Config kubernetesConfig, String oapiVersion, String openS
...
+        try {
+          this.openShiftUrl = new URL(new URL(getMasterUrl()), "/oapi/" + this.oapiVersion + "/").toString();
+        } catch (MalformedURLException e) {
+          throw KubernetesClientException.launderThrowable(e);
+        }
       }
     }
   }
 
+  public static String getCutsomOpenshiftUrl() {
```
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftConfig.java]: There's already a util for this, although can't remember where without looking.

File: openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftConfig.java [hunk 3]
```diff
@@ -58,15 +61,22 @@ public OpenshiftConfig(Config kubernetesConfig, String oapiVersion, String openS
...
+        try {
+          this.openShiftUrl = new URL(new URL(getMasterUrl()), "/oapi/" + this.oapiVersion + "/").toString();
+        } catch (MalformedURLException e) {
+          throw KubernetesClientException.launderThrowable(e);
+        }
       }
     }
   }
 
+  public static String getCustomOpenshiftUrl() {
```
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftConfig.java]: Can't just use Utils.getSystemPropertyOrEnvVar for this

File: openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftConfig.java (hunk 2 above)
Comment (iocanel) [inline openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftConfig.java]: found it

**Comment by iocanel** [comment]
[merge] fast before Jimmy gets back with an other comment!

**Pull request merged**